Logstash/logstash-input-azureblob/lib/logstash/inputs/azureblob.rb

```diff
@@ -135,6 +206,47 @@ def process(queue)
       @logger.error("Oh My, An error occurred. \nError:#{e}:\nTrace:\n#{e.backtrace}", :exception => e)
     end # begin
   end # process
+  
+  # Get first json object out of a string, return the rest of the string
+  def get_jsons(content, batch_size)
+    return nil, content, 0 if content.nil? || content.length == 0
+    return nil, content, 0 if (content.index '{').nil?
+
+    hit = 0
+    count = 0
+    index = 0
+    first = content.index('{')
+    move_opening = true
+    move_closing = true
+    while(hit < batch_size)
+        inIndex = content.index('{', index) if move_opening
+        outIndex = content.index('}', index) if move_closing
+
+        # TODO: Fix the ending condition
+        break if count == 0 && (inIndex.nil? || outIndex.nil?)
+        
+        if(inIndex.nil?)
+            index = outIndex
+        elsif(outIndex.nil?)
+            index = inIndex
+        else
+            index = [inIndex, outIndex].min
+        end #if
+        if content[index] == '{'
+            count += 1
+            move_opening = true
+            move_closing = false
+        elsif content[index] == '}'
+            count -= 1
+            move_closing = true
+            move_opening = false
```

Comment:
I had a really hard time trying to understand the logic for this function, and I spent 15 min starring at this function.
The nonconventional search of '{' and '}', that doesn't simply go character by character incrementally, was what made it hard to follow.

I am not sure if it would be more efficient if you loop through each character in Ruby yourself. It would be really good to add comments explaining your logic here, such as how you are search the '{' and '}', how the min function on line 233 allows you to still count the braces sequentially, and count is really for balancing the open and close brace, etc. More descriptive variable names would help a lot too.

Reply:
Having said that, I think your parsing logic can still fail if the literal string has a brace inside it. See if there are existing JSON parser that can do the job.

Reply:
For question 1, I benchmarked the performance, the current implementation is around 30 times faster than going through the string 1 by another character in typical cases.

That said, it is a good idea to introduce native json parser, the current catch is that json parser usually asks a well formed json. I'll keep looking into it.